### ISSUE TITLE
SPI WriteRead

### ISSUE THREAD

**Issue by joshkeys19** [Brilliant-Labs/bBoardFirmware-Revision1.1#18]
Add ability to perform SPI write-read

**Comment by joshkeys19** [comment]
Hi Josh,

Here's the updated code that include the SPI Write_Read function Id (0x06).  Simply send the module id, function id, and all the bytes that you want to write on the spi and the click board will send back (to the bboard Tx Buffer) the same number of read bytes it received during the write execution.  Keep in mind that the click board is currently limited to transmitting back up to 1024 bytes (as defined by MAX_CLICKBOARD_TX_DATA_BUFFER_SIZE) but that the message the clickboard receives from the bboard has some overhead of 2 bytes (module id and function id) therefore the max number of bytes you can write is 1024 -2 (where 1024 is defined by MAX_CLICKBOARD_RX_DATA_BUFFER_SIZE).

TLDR; SPI_BULK_WRITE_READ is implemented.  Current limitation is 1022 bytes.

Yves
